crates/solidity/outputs/npm/tests/src/compilation/binding-graph.test.mts

```diff
@@ -0,0 +1,125 @@
+    const reference = unit.bindingGraph.referenceAt(cursor)!;
+    assertUserFileLocation(reference.location, "child.sol", TerminalKind.Identifier, 3);
+
+    const defs = reference.definitions();
+    assert.equal(defs.length, 1);
+
+    assertUserFileLocation(defs[0]!.nameLocation, "parent.sol", TerminalKind.Identifier, 3);
+    assertUserFileLocation(defs[0]!.definiensLocation, "parent.sol", NonterminalKind.ContractDefinition, 2);
+  }
+
+  // contract Child is Parent {
+  //          ^^^^^
+  assert(cursor.goToNextTerminalWithKind(TerminalKind.Identifier));
+  assert.equal(cursor.node.unparse(), "Child");
+
+  {
+    const definition = unit.bindingGraph.definitionAt(cursor)!;
+    assertUserFileLocation(definition.nameLocation, "child.sol", TerminalKind.Identifier, 5);
+    assertUserFileLocation(definition.definiensLocation, "child.sol", NonterminalKind.ContractDefinition, 4);
+  }
+
+  {
+    assert.equal(unit.bindingGraph.referenceAt(cursor), undefined);
+  }
+
+  // contract Child is Parent {
+  //                   ^^^^^^
+  assert(cursor.goToNextTerminalWithKind(TerminalKind.Identifier));
+  assert.equal(cursor.node.unparse(), "Parent");
+
+  {
+    assert.equal(unit.bindingGraph.definitionAt(cursor), undefined);
+  }
+
+  {
+    const reference = unit.bindingGraph.referenceAt(cursor)!;
+    assertUserFileLocation(reference.location, "child.sol", TerminalKind.Identifier, 5);
+
+    const defs = reference.definitions();
+    assert.equal(defs.length, 2);
+
+    assertUserFileLocation(defs[0]!.nameLocation, "child.sol", TerminalKind.Identifier, 3);
+    assertUserFileLocation(defs[0]!.definiensLocation, "child.sol", NonterminalKind.ImportDeconstructionSymbol, 3);
+
+    assertUserFileLocation(defs[1]!.nameLocation, "parent.sol", TerminalKind.Identifier, 3);
+    assertUserFileLocation(defs[1]!.definiensLocation, "parent.sol", NonterminalKind.ContractDefinition, 2);
+  }
+
+  //   function foo() public pure {
+  //            ^^^
+  assert(cursor.goToNextTerminalWithKind(TerminalKind.Identifier));
+  assert.equal(cursor.node.unparse(), "foo");
+
+  {
+    const definition = unit.bindingGraph.definitionAt(cursor)!;
+    assertUserFileLocation(definition.nameLocation, "child.sol", TerminalKind.Identifier, 6);
+    assertUserFileLocation(definition.definiensLocation, "child.sol", NonterminalKind.FunctionDefinition, 6);
+  }
+
+  {
+    assert.equal(unit.bindingGraph.referenceAt(cursor), undefined);
+  }
+
+  //     assert(true);
+  //     ^^^^^^
+  assert(cursor.goToNextTerminalWithKind(TerminalKind.Identifier));
+  assert.equal(cursor.node.unparse(), "assert");
+
+  {
+    assert.equal(unit.bindingGraph.definitionAt(cursor), undefined);
+  }
+
+  {
+    const reference = unit.bindingGraph.referenceAt(cursor)!;
+    assertUserFileLocation(reference.location, "child.sol", TerminalKind.Identifier, 7);
+
+    const defs = reference.definitions();
+    assert.equal(defs.length, 0);
+    // TODO: there should be one built-in definition, right?
```

Comment:
Correct. This is probably because there's a missing step when constructing the `BindingsGraph` in `CompilationUnit` to add the built-ins. Since this is language dependant, I don't think it's possible to do in the runtime. As of today's base `main` this is also a bit cumbersome, but I streamlined the process in #1170. It may be worth it to wait for it to be merged (shouldn't be too far ahead) to find a good solution for this.

Reply:
I made the refactoring, and added a couple of `TODO`s that you might want to check for the remaining work. I didn't want to do a larger refactoring of bindings until you get a chance to review/rebase on top.